src/components/ng2-map.component.ts

```diff
@@ -114,7 +114,7 @@ export class Ng2MapComponent implements OnChanges, OnDestroy, AfterViewInit {
       window['ng2MapRef'].forEach( ng2MapRef => {
         ng2MapRef.zone.run(function() { ng2MapRef.componentFn(); });
       });
-      window['ng2MapRef'] = [];
+      window['ng2MapRef'].splice(0, window['ng2MapRef'].length);
```

Comment:
I see, thanks for the explanation. Also, it's good to move those to Ng2Map service. However, before you do action and make PR, please create an issue first, so that other members can see what's coming in the future and to avoid duplicate work.